minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java

```diff
@@ -468,6 +469,16 @@ protected void setHostname() {
   protected void setAlluxioHome() throws IOException {
     mHome =
         File.createTempFile("Alluxio", "U" + System.currentTimeMillis()).getAbsolutePath();
+    // Make a copy of mHome.
+    final String home = mHome;
+    Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
```

Comment:
A: To my knowledge it's fine - do you have a particular concern?

We can't use junit temporary dir because this directory is shared between tests and even between test clusters. It's bad that it's shared in this way, but untangling that issue is a much bigger challenge. This change is minimally invasive while vastly improving our test cleanliness.